SwiftTweaks/Tweak.swift

```diff
@@ -88,16 +88,15 @@ extension Tweak where T: Comparable {
 	}
 }
 
-extension Tweak {
-	// This can't be an initializer because generic types apparently can't handle fixed-type initializers
-	public static func stringList(_ collectionName: String, _ groupName: String, _ tweakName: String, options: [String], defaultValue: String?=nil) -> Tweak<StringOption> {
+extension Tweak where T == StringOption {
+	public init(_ collectionName: String, _ groupName: String, _ tweakName: String, options: [String], defaultValue: String? = nil) {
 		precondition(!options.isEmpty, "Options list cannot be empty (stringList tweak \"\(tweakName)\")")
 		precondition(
-			defaultValue == nil || (defaultValue != nil && options.index(of: defaultValue!) != nil),
+			defaultValue == nil || (defaultValue != nil && options.contains(defaultValue!) == false),
```

Comment:
I think there's a bug here, right?

The pseudocode for what we want is:

> "Either the default value is nil,
> OR
> if there is a default value, make sure it's contained in `options`"
> 

The second part of this precondition, as written here, is `"if there is a default value, make sure it's *not* contained in `options`")

so this line of code should be:
`precodnition(defaultValue == nil || (defaultValue != nil && options.contains(defaultValue!))`

Reply:
...or even-more concisely:

`precondition(defaultValue == nil || options.contains(defaultValue!))`

Reply:
Yeah! You're right!
Thanks for pointing it out.